### PR TITLE
Feature/estimate request

### PR DESCRIPTION
**Pull request by riruADS**

### CODE REVIEW

File: app/javascript/bundles/Admin/components/EstimateRequests/TableBody.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Link } from 'react-router-dom'
```
Comment (Skumring) [inline app/javascript/bundles/Admin/components/EstimateRequests/TableBody.jsx]: looks like we don't need import the `Link` (not used anywhere)

File: app/assets/javascripts/beapp/estimate_request.js.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// File loader
+var fileInput = document.getElementById('estimate_request_document');
+var filenameContainer = document.getElementById('click-area');
+
+if (fileInput) {
+  fileInput.addEventListener('change', function() {
+    filenameContainer.innerText = fileInput.value.split('\\').pop() || 'Click to load your file ...';
```
Comment (Skumring) [inline app/assets/javascripts/beapp/estimate_request.js.erb]: looks like you don't check presence of `filenameContainer`

File: app/assets/javascripts/beapp/application.js.erb [hunk 3]
```diff
@@ -7,6 +7,8 @@
 //= require beapp/jquery.counterup
 //= require beapp/jquery.magnific-popup
 //= require beapp/imgfix
+//= require beapp/estimate_request
```
Comment (skliarov) [inline app/assets/javascripts/beapp/application.js.erb]: @riruADS this should be our own JS, not beapp...

File: app/assets/stylesheets/beapp/_estimate_request.scss [hunk 4]
```diff
@@ -0,0 +1,128 @@
+/*
```
Comment (skliarov) [inline app/assets/stylesheets/beapp/_estimate_request.scss]: @riruADS same here, it's part of the `app/...`, not `beapp/...`

File: app/controllers/estimate_requests_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+class EstimateRequestsController < ApplicationController
+  # GET /request-estimate
```
Comment (skliarov) [inline app/controllers/estimate_requests_controller.rb]: @riruADS are you sure this comment reflects the request information?

File: app/serializers/article_index_serializer.rb [hunk 6]
```diff
@@ -1,15 +0,0 @@
-class ArticleIndexSerializer < ActiveModel::Serializer
```
Comment (skliarov) [inline app/serializers/article_index_serializer.rb]: @riruADS why did you delete this one?

File: config/environments/development.rb [hunk 7]
```diff
@@ -49,5 +52,6 @@
 
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
-  config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  # config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+  config.active_job.queue_adapter = :delayed_job
```
Comment (skliarov) [inline config/environments/development.rb]: @riruADS it was enough to list it in application.rb

File: config/environments/production.rb [hunk 8]
```diff
@@ -74,4 +74,6 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+  
+  config.active_job.queue_adapter = :delayed_job
```
Comment (skliarov) [inline config/environments/production.rb]: @riruADS it was enough to list it in application.rb

File: app/jobs/recurring/send_beacon_email_job.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+class SendBeaconEmailJob
+  include Delayed::RecurringJob
+  
+  run_every 1.day
+  run_at '06:00am'
```
Comment (skliarov) [inline app/jobs/recurring/send_beacon_email_job.rb]: @riruADS why didn't you set a queue for this job?

File: app/controllers/estimate_requests_controller.rb (hunk 5 above)
Comment (skliarov) [reply app/controllers/estimate_requests_controller.rb]: Nevermind, it does.

File: Gemfile [hunk 10]
```diff
@@ -57,8 +58,15 @@ gem 'mini_magick'
 # AWS S3 adapter for Carrierwave
 gem 'fog-aws'
 
+# Background jobs
+gem 'delayed_job_active_record'
+gem 'delayed_job_recurring'
```
Comment (skliarov) [inline Gemfile]: @riruADS you forgot to install daemons for running delayed_job on system level.
